python/prairielearn.py

```diff
@@ -125,23 +159,33 @@ def from_json(v):
                 if ('_value' in v) and ('real' in v['_value']) and ('imag' in v['_value']):
                     return complex(v['_value']['real'], v['_value']['imag'])
                 else:
-                    raise Exception('variable of type complex should have value with real and imaginary pair')
+                    raise ValueError('variable of type complex should have value with real and imaginary pair')
```

Comment:
I don't think there is a way to incorporate a more specific error message here related to the problematic variable. e.g. 

```python
ValueError(f'Variable {variable_name} with type complex should have both with real and imaginary pair values inside of value.')
```

`variable_name` I think has to be passed in as there doesn't seem to be a good option to extract the information via `inspect`.

Reply:
Yeah, this is kindof a persistent issue unfortunately. Debugging these json serialization issues is always a pain.

Regarding your other comment, since it seems like we might go in another direction with the dataframes, I'll remove those changes from this PR and flip it to open (should get merged sooner that way).

Reply:
I thought about this a bit over break. I've surmised my thoughts here: 

https://github.com/PrairieLearn/PrairieLearn/issues/6883

Maybe we could have this as a _separate_ PR to follow once the updated JSON conversions are included.